controllers/AuthController.js

```diff
@@ -1,69 +1,87 @@
-            console.log('User added successfully!');
-        })
-        .catch(error => {
-            res.json({
-                message: 'An error occured'
-            })
-        })
-    })
-}
+  bcrypt.hash(req.body.password, 10, function (err, hashedPass) {
+    if (err) {
+      res.json({
+        error: err,
+      });
+    }
+    let user = new User({
+      fname: req.body.fname,
+      lname: req.body.lname,
+      email: req.body.email,
+      password: hashedPass,
+      phone: req.body.Phonenumber,
+    });
 
-const login = (req, res, next) =>{
-    var username = req.body.username
-    var password = req.body.password
+    user
+      .save()
+      .then((user) => {
+        res.redirect("/");
+        console.log("User added successfully!");
+      })
+      .catch((error) => {
+        res.json({
+          message: "An error occured",
+        });
+      });
+  });
+};
 
-    User.findOne({$or: [{email:username}]})
-    .then(user => {
-        if(user){
-            bcrypt.compare(password, user.password, function(err, result){
-                if(err){
-                    res.join({
-                        error: err
-                    })
-                }
-                if(result){
-                    let token = jwt.sign({name: user.name}, 'verySecretValue', {expiresIn: '1h'})
-                    //res.redirect('/home');
-                    res.sendFile(path.join(__dirname ,  '../feed_2.0.html'));
-                    console.log('Logged in successfully!');
-                }else{
-                    res.json({
-                        message: 'Password did not match!'
-                    })
-                }
-            })
-        }else{
-            res.json({
-                message: 'No user found!'
-            })
+const login = (req, res, next) => {
+  var username = req.body.username;
+  var password = req.body.password;
+
+  User.findOne({ $or: [{ email: username }] }).then((user) => {
+    if (user) {
+      bcrypt.compare(password, user.password, function (err, result) {
+        if (err) {
+          res.join({
+            error: err,
+          });
+        }
+        if (result) {
+          /*let token = jwt.sign({ name: user.name }, "verySecretValue", {
+            expiresIn: "1h",
+          });*/
+          const token = createToken(user._id);
+          res.cookie("jwt", token, { httpOnly: true, maxAge: 1000 * 1000 });
+          res.redirect("/feed");
+          console.log("Logged in successfully!");
+          console.log(token);
+        } else {
+          res.json({
+            message: "Password did not match!",
+          });
         }
-    })
-}
+      });
+    } else {
+      res.json({
+        message: "No user found!",
+      });
+    }
+  });
+};
+
+const logout = (req, res, next) => {
+  res.cookie("jwt", "", { maxAge: 1 });
+  res.redirect("/");
+};
 
 module.exports = {
-    register, login
-}
\ No newline at end of file
+  register,
+  login,
+  logout,
+};
```

Comment:
in regards to Auth, we have to find a cleaner way to authenticate the backend without the need for a frontend.